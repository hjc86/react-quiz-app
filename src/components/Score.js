import React from 'react';
import styles from './Score.module.css';
import ScoreBoard from './ScoreBoard';
import ScoreCount from './ScoreCount';


class Score extends React.Component{
    
    constructor(props){

        super(props)

        this.state={
            winner: " "
        }
    }
    
    checkWinner = () => {

        let winner = this.props.scoreArray.indexOf(Math.max(...this.props.scoreArray));

       if (winner === 0) {
            this.setState({
                winner: this.props.userName[0]
                })
        } else if (winner === 1) {
            this.setState({
                winner: this.props.userName[1]
                })
        } else if (winner === 2) {
            this.setState({
                winner: this.props.userName[2]
                })
        } else if (winner === 3) {
            this.setState({
                winner: this.props.userName[3]
                })
        } else {
            this.setState({
                winner: this.props.userName[4]
                })
        }
    }


    // getPlayerItems = () => { 
    //     this.props.userName.map((userNames)=> 
    //         userNames);
    //     return (
    //         this.userNames
    //     )
    // }

   
    componentDidMount(){
        console.log("score updated ")
        this.checkWinner();
    }

    refreshPage = () => {
        window.location.reload(false);
      }

    render() {


        return(
            <div className={styles.scoreComponent}> 
                <div className={styles.scoreComponentResults}>
                    <h1 className={styles.scoreResultsChild}>The winner is:{this.state.winner} </h1>
                    
                    <h3 className={styles.scoreResultsChild}>Scoreboard</h3>
                    <div className={styles.scoreGroup}>
                        <div className={styles.scoreBoard}>  
                            {this.props.userName.map((scoreBoard)=> (
                            <ScoreBoard {...this.state} scoreBoard={scoreBoard}/>
                        ))}
                        </div>
                        <div className={styles.scoreCount}>
                        {this.props.scoreArray.map((scoreCount)=> (
                            <ScoreCount {...this.state} scoreCount={scoreCount} />
                        ))}
                        </div>
                    </div>

                   {/* <div className="scoreBoard">
                       <ul className={styles.scoreResultsChild}>Scoreboard:
                          <li className={styles.scoreResultsChild}>{this.props.userName[0]} : {this.props.scoreArray[0]}</li> 
                          <li className={styles.scoreResultsChild}>{this.props.userName[1]} : {this.props.scoreArray[1]}</li> 
                          <li className={styles.scoreResultsChild}>{this.props.userName[2]} : {this.props.scoreArray[2]}</li> 
                          <li className={styles.scoreResultsChild}>{this.props.userName[3]} : {this.props.scoreArray[3]}</li> 
                          <li className={styles.scoreResultsChild}>{this.props.userName[4]} : {this.props.scoreArray[4]}</li> 
                        </ul>
                   </div> */}

                    
                </div>

                <div className={styles.skyContainer}>
                    <div className={styles.star}></div>
                    <div className={styles.star}></div>
                    <div className={styles.star}></div>
                    <div className={styles.star}></div>
                    <div className={styles.star} ></div>
                </div>

                

                <div className={styles.startButtonDiv}>
                        <button className={styles.startButton} onClick={this.refreshPage}>Start New Quiz!</button>
                </div>

                
                
            </div>
            
        
        )
    }
}

export default Score;

