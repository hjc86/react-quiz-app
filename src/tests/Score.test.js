import React, { Component } from 'react';
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme';
import Score from '../components/Score';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';


describe("<Score />", () => {

    it('mounts to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Score />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<Score />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    let wrapper;
    beforeEach(() => wrapper = shallow(<Score />));
    
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render a <div />', () => {
        wrapper = shallow(<Score />)
        expect(wrapper.find('div').length).toEqual(1);
    });

   it('should render two <p />', () => {
        expect(wrapper.find('p').length).toEqual(2);
    });



})