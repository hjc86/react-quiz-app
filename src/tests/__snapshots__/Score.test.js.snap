// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score /> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<Score /> renders correctly 1`] = `
<div
  className="scoreComponent"
>
  <div
    className="scoreComponentResults"
  >
    <h1
      className="scoreResultsChild"
    >
      Score Board 
    </h1>
    <p
      className="scoreResultsChild"
    >
      a
       scored 
      0
      /5!
    </p>
    <p
      className="scoreResultsChild"
    >
      b
       scored 
      1
      /5!
    </p>
  </div>
  <div
    className="skyContainer"
  >
    <div
      className="star"
    />
    <div
      className="star"
    />
    <div
      className="star"
    />
    <div
      className="star"
    />
    <div
      className="star"
    />
  </div>
  <div
    className="startButtonDiv"
  >
    <button
      className="startButton"
      onClick={[Function]}
    >
      Play Again
    </button>
  </div>
</div>
`;
